mvc/views/razor.rst

```diff
@@ -1,8 +1,283 @@
+html tags and the ``<text>`` tag. The ``<text>`` tag is a placeholder tag that
+is only used server side to tell razor that the following content is text and
+should not be parsed as c# code.
+
+.. code-block:: c#
+
+  @{
+    var title = "Hello!";
+    <h1>@title</h1>
+
+    var body = "some text not sitting inside of a tag";
+    <text>@body</text>
+  }
+
+*******************
+Control Structures
+*******************
+A lot of the power behind razor comes from the ability to have control structures
+inside of your markup. C# control structures such as ``foreach``, ``if``,
+``using``, etc are available with Razor.
+
+.. code-block:: c#
+
+  @if(Model.IsRenderable){
+    <ul>
+    @foreach(var item in Model.Items){
+      <li>@item.Name</li>
+    }
+    </ul>
+  }
+
+
+*********
+Comments
+*********
+The Razor ``@* *@`` comment block is analogous to the c# ``/* */`` comment
+block:
+
+.. code-block:: c#
+
+  @*
+    Some comment
+  *@
+
+***************************
+Application Relative Paths
+***************************
+One of my favorite features of razor is the ability to use application relative
+paths.
+
+.. code-block:: c#
+
+  <a href="~/Home">Home page</a>
+  <script type='text/javascript' src="~/some.js">
+
+This only works for attributes that are defined in HTML5 as urls. If you want
+to resolve a path in an attribute not defined as a url you can use
+
+.. code-block:: c#
+
+  <a cust-attr="@Url.Content("~/the-url")" href="#"> Link </a>
+
+***************
+Tag Attributes
+***************
+Razor makes it easy to have programmatic classes and ids for you html tags.
+
+.. code-block:: c#
+
+  @{
+    var AnchorClass = "dark link";
+    var AnchorId = "main-link";
+  }
+
+  <a href="#" class="@AnchorClass" id="@AnchorId"> Link </a>
+
+The output for this markup is below.
+
+.. code-block:: html
+
+  <a href="#" class="dark link" id="main-link"> Link </a>
+
+When the value you pass into an attribute is null razor will remove that
+attribute completly.
+
+.. code-block:: c#
+
+  @{
+    string classes = null;
+  }
+
+  <a href="#" class="@classes"> Link </a>
+
+.. code-block:: Html
+
+  <a href="#"> Link </a>
+
+If you want the attribute to exist, but to be empty then you use an empty string.
+
+.. code-block:: c#
+
+  @{
+    string classes = "";
+  }
+
+  <a href="#" class="@classes"> Link </a>
+
+.. code-block:: Html
+
+  <a href="#" class=""> Link </a>
+
+It doesn't just work with strings, but boolean values as well. This is useful
+for programatically disabling input fields, checking checkboxes, etc.
+
+.. code-block:: c#
+
+  @{
+    bool agree = true;
+    bool enabled = false;
+  }
+
+  <input type="checkbox" name="agree" value="agree" checked="@agree"/>
+  <input type="checkbox" name="option" value="option" disabled="@enabled"/>
+
+.. code-block:: html
+
+  <input type="checkbox" name="agree" value="agree" checked="checked"/>
+  <input type="checkbox" name="option" value="option" disabled="disabled"/>
+
```

Comment:
Should add 1 more example showcasing that you don't get the magic `true`/`false` behavior when there are multiple dynamic expressions in an attribute:

`<input custom="@true hi" />` => `<input custom="True hi" />`
`<input custom="@true hi @null" />` => `<input custom="True hi" />` Note the missing space after `hi`
`<input custom="@null @null" />` => `<input custom="" />`